### PR TITLE
More CGU cleanups

### PR DESCRIPTION
**Pull request by nnethercote**
An assortment of improvements.

r? @wesleywiser

### CODE REVIEW

**Comment by nnethercote** [comment]
Should have no perf effect, let's check:

@bors try @rust-timer queue

**Comment by Kobzol** [comment]
`coercions debug incr-full` has become very noisy recently, so this is just noise.

**Review by wesleywiser — APPROVED** [review]
r=me with or without nit 🙂

File: compiler/rustc_monomorphize/src/partitioning.rs [hunk 1]
```diff
@@ -512,21 +498,24 @@ fn internalize_symbols<'tcx>(
...
-                        // instantiated. We can safely ignore those.
-                        mono_item_placements.get(user_item)
-                    })
-                    .any(|placement| *placement != home_cgu)
-                {
-                    // Found a user from another CGU, so skip to the next item
-                    // without marking this one as internal.
-                    continue;
+
+            if !single_codegen_unit {
```
[the remark this answers is not in the document]
Comment (nnethercote) [reply compiler/rustc_monomorphize/src/partitioning.rs:486]: I felt like the "Found a user from another CGU" comment below covered the reasoning sufficiently here.

**Pull request merged**